packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -3331,6 +3331,7 @@ class RenderSemanticsGestureHandler extends RenderProxyBox {
     GestureDragUpdateCallback onHorizontalDragUpdate,
     GestureDragUpdateCallback onVerticalDragUpdate,
     this.scrollFactor = 0.8,
+    this.semanticsConfigurationOverride,
```

Comment:
It seems odd to pass the SemanticsConfiguration object to the widget layer. It's really just a rendering layer concept.

Would the following work:
* We extend the RenderSemanticsGestureHandler to also allow specifiying "onScrollLeft", "onScrollRight", "onScrollUp", "onScrollDown".
* `RawGestureDetector.semantics` already allows you to customize the semantics of the RenderSemanticsGestureHandler, it could also provide a custom onScrollLeft, onScrollRight, etc.
* We need to pipe `RawGestureDetector.semantics` through `Scrollable` to ListView/PageView so it can be customized from there.